client.go

```diff
@@ -0,0 +1,130 @@
+package ndp
+
+import (
+	"fmt"
+	"net"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+)
+
+// A Conn is a Neighbor Discovery Protocol connection.
+type Conn struct {
+	pc *ipv6.PacketConn
+	cm *ipv6.ControlMessage
+
+	allNodes *net.IPAddr
+}
+
+// Dial dials a NDP connection using the specified interface.  It returns
+// a Conn and the link-local IPv6 address of the interface.
```

Comment:
Why does the caller need to know the lladdr? From my reading of NDP, it's only needed to source the packets, so could stay internal to this package?

Reply:
It's useful, at least for now, until we have more specialized methods available.  Then I can consider dropping it.